File: Hangman.py
def Hangman(word):
  #All the varibles and lists
  letters = []
  lives = 9
  game_over = False
  game_won = False
  letters_guessed = []
  spaced_letters = []
  guessed = 0
  word.lower()
  
  print("Hello! Welcome to Hangman.")
  
  #Making the word lists
  for i in range(len(word)):
    letters.append(word[i])
  for i in letters:
      spaced_letters.append("_")
    
  while game_over == False or game_won == False:
    guessed = 0
    
    #Start of each round
    for i in spaced_letters:
      print(i, end = " ")
    print("\nYou have " + str(lives) + " lives.")
    guess = input("Guess: ")
    
    #Check if the guessed letter is correct or incorrect
    if guess in letters:
      if guess in spaced_letters:
        print("You already guessed this letter correctly.")
      else: 
        print(guess, "is in the word.")
        for i in range(len(letters)):
          if letters[i] == guess:
            spaced_letters[i] = guess;
    elif guess.isalpha() == False or len(guess) > 1:
      print("Make sure to put one letter only.")
    else:
      if guess not in letters_guessed:
        letters_guessed.append(guess)
        print("Sorry! That is not in the word.")
        lives -= 1
      else: 
        print("You already guessed this")
        
    #Checking if the player has won or lost in this round   
    for i in range(len(letters)):
      if letters[i] == spaced_letters[i]:
        guessed += 1
    if guessed == len(letters):
      return print("YOU WON!! The word was", word)
      game_won = True
    if lives == 0:
      return print("YOU LOST!! The word was", word)
      game_over = True
      
print(Hangman("Plane"))

File: Hangman.java
import java.util.*;

public class Hangman{
  private int lives;
  private String word;
  private ArrayList<String> lettersInWord = new ArrayList<String>();
  private ArrayList<String> lettersGuessed = new ArrayList<String>();
  private ArrayList<String> displayWord = new ArrayList<String>();
  private Scanner scanner;
  private Boolean won;
  
  public Hangman(String wordToGuess){
    word = wordToGuess;
    lives = 9;
    won = false;
    scanner = new Scanner(System.in);
    for(int i = 0;i < word.length(); i++){
      lettersInWord.add(word.substring(i, i+1));
      displayWord.add("_");
    }
  }

  public void round(){
    while(lives > 0 && displayWord.contains("_")){
      System.out.println("You have " + lives + " lives left.");
      System.out.print("Current Word: ");
      for(String i : displayWord){
        System.out.print(i + " ");
      }
      System.out.println(" ");
      System.out.print("Enter your guess: ");
      String guess = scanner.nextLine();
      boolean works = checkAns(guess);
      
      if(works){
        boolean guessed = false;
        for(int j = 0; j < lettersInWord.size(); j++){
          if(guess.equals(lettersInWord.get(j))){
            displayWord.set(j, guess);
            guessed = true;
          }
        }
        if(guessed == true){
          System.out.println("You guessed a letter!");
        }
        else{
          System.out.println("You didn't guess any letters.");
          lives--;
        }
        
        System.out.println(" ");
        lettersGuessed.add(guess);
        }
      }
    
      end();
  }

  public boolean checkAns(String guess){
      if(lettersGuessed.contains(guess)){
        System.out.println("You already guessed this letter. Guess another letter. :)");
        System.out.println(" ");
        return false;
      }
      return guess.length() == 1 && Character.isLetter(guess.charAt(0));
    }
  
  public void end(){
    if(lettersInWord.equals(displayWord)){
      System.out.println("You've guessed the word. Congrats!!");
    }
    else{
      System.out.println("You didn't guess the word. It was " + word + " :(");
    }
  }
}
